File: Examples/CSharp/Articles/StylingAndDataFormatting/ExtractThemeData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Cells.Examples.CSharp.Articles.StylingAndDataFormatting
{
    public class ExtractThemeData
    {
        public static void Run()
        {
            //Source directory
            string sourceDir = RunExamples.Get_SourceDirectory();

            // Create workbook object
            Workbook workbook = new Workbook(sourceDir + "sampleExtractThemeData.xlsx");

            // Extract theme name applied to this workbook
            Console.WriteLine(workbook.Theme);

            // Access first worksheet
            Worksheet worksheet = workbook.Worksheets[0];

            // Access cell A1
            Cell cell = worksheet.Cells["A1"];

            // Get the style object
            Style style = cell.GetStyle();

            if (style.ForegroundThemeColor != null)
            {
                // Extract theme color applied to this cell if theme has foregroundtheme color defined
                Console.WriteLine(style.ForegroundThemeColor.ColorType);
            }
            else
            {
                Console.WriteLine("Theme has no Foreground Color defined.");
            }

            // Extract theme color applied to the bottom border of the cell if theme has border color defined
            Border bot = style.Borders[BorderType.BottomBorder];
            if (bot.ThemeColor != null)
            {
                Console.WriteLine(bot.ThemeColor.ColorType);
            }
            else
            {
                Console.WriteLine("Theme has no Border Color defined.");
            }

            Console.WriteLine("ExtractThemeData executed successfully.");
        }
    }
}

File: Examples/VisualBasic/Articles/StylingAndDataFormatting/ExtractThemeData.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace Articles.StylingAndDataFormatting
    Public Class ExtractThemeData
        Public Shared Sub Run()
            ' ExStart:1
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            ' Create workbook object
            Dim workbook As New Workbook(dataDir & Convert.ToString("source.xlsx"))

            ' Extract theme name applied to this workbook
            Console.WriteLine(workbook.Theme)

            ' Access first worksheet
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Access cell A1
            Dim cell As Cell = worksheet.Cells("A1")

            ' Get the style object
            Dim style As Style = cell.GetStyle()

            If style.ForegroundThemeColor IsNot Nothing Then
                ' Extract theme color applied to this cell if theme has foregroundtheme color defined
                Console.WriteLine(style.ForegroundThemeColor.ColorType)
            Else
                Console.WriteLine("Theme has not foreground color defined.")
            End If

            ' Extract theme color applied to the bottom border of the cell if theme has border color defined
            Dim bot As Border = style.Borders(BorderType.BottomBorder)
            If bot.ThemeColor IsNot Nothing Then
                Console.WriteLine(bot.ThemeColor.ColorType)
            Else
                Console.WriteLine("Theme has not Border color defined.")
            End If
            ' ExEnd:1
        End Sub
    End Class
End Namespace